### PR TITLE
Sending palette changes to pangolin laser via OSC

### PR DESCRIPTION
**Pull request by worldveil**
Adding a loop task for the LX engine to send OSC messages with the current palette (primary) color to the lasers. 

Message will look like the following: `/lx/color/hue < int in range [0, 360) >;`

### CODE REVIEW

**Review by jkbelcher — COMMENTED** [review]
The conversion to LXLoopTask is solid.  Added a question about taking saturation/brightness into consideration for the color info sent to lasers.

Added comments about an `enabled` switch.  I think it's a quick add to do it now but you can defer to later if you want.

File: src/main/java/titanicsend/lasercontrol/TELaserTask.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    @Override
+    public void loop(double deltaMs) {
+        // get the swatch color
+        int primaryIndex = TEPattern.ColorType.PRIMARY.swatchIndex();
+        LXDynamicColor primary = this.lx.engine.palette.swatch.getColor(primaryIndex);
+
+        // convert to a 0 - 360 format for Pangolin
+        int hue = (int)(primary.getHue());
```
[the remark this answers is not in the document]
Comment (worldveil) [reply src/main/java/titanicsend/lasercontrol/TELaserTask.java:24]: Not sure without Pangolin in front of me, unfortunately. Perhaps @jvyduna knows. I do know we got RGB working on 0-360 scale though.

**Comment by worldveil** [comment]
@jkbelcher ok added the flag and inheritance for now - can do UI work later

File: src/main/java/titanicsend/lasercontrol/TELaserTask.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jkbelcher) [reply src/main/java/titanicsend/lasercontrol/TELaserTask.java]: Ok if 0-360 is working we'll start with that and expand if/when Jeff gets back to us.

**Review by jkbelcher — APPROVED** [review]
Made a couple adjustments to the LXComponent setup.  By using a BooleanParameter instead of boolean the state will get saved/loaded from the .lxf file.

If this looks good to you merge away.

**Pull request merged**